WooCommerce/Classes/POS/Presentation/CardReaderConnection/UI States/Reader Messages/PointOfSaleCardPresentPaymentProcessingMessageView.swift

```diff
@@ -6,11 +6,8 @@ struct PointOfSaleCardPresentPaymentProcessingMessageView: View {
 
     var body: some View {
         VStack(alignment: .center, spacing: Layout.headerSpacing) {
-            Image(decorative: viewModel.imageName)
-                .resizable()
-                .aspectRatio(contentMode: .fill)
-                .frame(width: PointOfSaleCardPresentPaymentLayout.headerSize.width,
-                       height: PointOfSaleCardPresentPaymentLayout.headerSize.height)
+            ProgressView()
+                .progressViewStyle(WaveProgressViewStyle())
```

Comment:
When we transition from `ReaderMessageView` to -> `PaymentProcesingMessageView` the progress view seems to "blink" since the previous view fades out and the next one fades in. It happens due to the default `transition(.opacity)` applied to all the cardReaderViews. 

We could fix this by explicitly setting a different transition for this view:

```
        .padding(.bottom)
        .multilineTextAlignment(.center)
        .transition(.asymmetric(insertion: .identity, removal: .opacity)) // no transition when inserting, fades out on removal
```

Result:

https://github.com/user-attachments/assets/94027f18-b141-4672-8c4e-1bf8d0d223b9

We don't have a mechanism implemented to dictate explicitly that this transition needs to be applied when going from one state to another. We could consider implementing such a mechanism in the future.

Reply:
Nice, thanks.

It would be even better if it could just be _the same_ progress view so that it just keeps going, but that doesn't fit too well with our view structure. With some of the suggested animations morphing from one into the other, we may have to deal with that kind of thing in future.